crates/proof-of-sql/src/tests/TestScript.t.sol

```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+library TestScript {
+    struct CustomStruct {
+        uint256 value;
+    }
+
+    error DummyError();
+
+    function rustTestWeCanThrowErrorDependingOnParameter(uint256 x) public pure {
+        if (x != 1234) revert DummyError();
+    }
+
+    function rustTestWeCanDecodeCustomStruct(bytes memory x) public pure {
+        (CustomStruct memory y) = abi.decode(x, (CustomStruct)); // solhint-disable-line no-unused-vars
```

Comment:
we can decode any struct or type but solidity should be aware of the type (CustomStruct here). unfortunately we can't pass the type as parameter via tx

Reply:
Renamed the method. The function is used to demonstrate that we can, in effect, pass a custom struct from Rust to Solidity. Didn't realize a tx couldn't handle a type as a parameter.